src/coresyf_image_crop/tests/test_image_crop.py

```diff
@@ -37,3 +36,19 @@ def test_get_shapefile_crs(self):
         epsg_code = get_shapefile_crs(self.data_source)
         self.assertEqual(epsg_code, self.epsg_code)
 
+    def test_apply_buffer_to_polygon(self):
+        buffer_crs = get_raster_crs(gdal.Open(self.test_image_4236))
+        polygon_extent = get_shapefile_polygon_extent(self.data_source)
+        buffer_4326 = 0.022207
+        buffer_3763 = 2000
+        polygon_crs = get_shapefile_crs(self.data_source)
+
+        polygon_with_buffer = apply_buffer_to_polygon(buffer_4326, polygon_extent,
+                                                      buffer_crs, polygon_crs)
+
+        expected_polygon = ogr.CreateGeometryFromWkt(self.extent_polygon_wkt)
+        expected_polygon = expected_polygon.Buffer(buffer_3763, 0)
+
+        for n in range(4):
+            self.assertAlmostEqual(polygon_with_buffer.GetEnvelope()[n],
```

Comment:
Why do I need to use an approximation? I think if we follow my previous suggestions we can use the (exact) equality comparison.